services/static.js

```diff
@@ -42,8 +42,11 @@ var Static = {};
  *   [`GeoJsonOverlay`](#geojsonoverlay)
  *
  * @param {boolean} [config.highRes=false]
- * @param {string} [config.insertOverlayBeforeLayer] - The ID of the style layer
+ * @param {string} [config.before_layer] - The ID of the style layer
  *   that overlays should be inserted *before*.
+ * @param {Object} [config.addlayer] - Adds a Mapbox style layer to the map's style at render time. Can be combined with before_layer.
+ * @param {Array} [config.setfilter] - Applies a filter to an existing layer in a style using Mapbox's expression syntax. Must be used with layer_id.
```

Comment:
I ran into some issues while testing this locally. The way we parse query params expects that any array is just a list of things you want to append as new parameters, so we just turn those arrays into a string of comma-separated values. This turns something like `[">", "height", 300]` into `>,height,300`, which makes the static API unhappy because it expects an actual array.

Reply:
@katydecorah i realized my above suggestion wasn't working, so I pushed a couple of other changes just for the sake of discussion. I'm sure there's a cleaner way to do this! https://github.com/mapbox/mapbox-sdk-js/pull/330/commits/85fd40f34c61d856898faa27b34489d13073c69f

Reply:
> I ran into some issues while testing this locally.

Do you have any example of a query that you ran into issues with locally?

Reply:
@katydecorah I just grabbed some examples from the tests: 

```
const staticReq = staticService.getStaticImage({
  ownerId: 'mapbox',
  styleId: 'streets-v11',
  width: 500,
  height: 600,
  position: {
    coordinates: [-74.01254,40.70577],
    zoom: 16
  },
  setfilter: [">","height",100],
  layer_id: 'building'
});
```

Which resulted in a request URL of:

```
https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/-74.01254,40.70577,16/500x600?setfilter=%3E%2Cheight%2C100&layer_id=building&access_token=
```

Reply:
Ohh I see! Hmm yes, v funky